site/book/07-effective-customizations/03-generation.md

```diff
@@ -0,0 +1,15 @@
+## Scenario:
+
+When using template solutions like Helm I am able to provide conditional statements based on parameter values.  This allows me to ask the user for a little bit of information and generate a lot of boilerplate configuration.
+
+## Problems:
+
+1. Over time the templating logic becomes its own language that becomes very complex.  Debugging the template generation becomes a task of its own.
```

Comment:
example:
https://github.com/nghnam/kube-common-setup/blob/master/templates/cluster-role-bindings.yaml

Reply:
thanks for linking to this, this is a good size package that we can possibly adopt since it's small and very readable.